assets/shared/structure/structure-store.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * External dependencies
+ */
+import { isEqual } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { apiFetch, controls as dataControls } from '@wordpress/data-controls';
+import { dispatch, registerStore, select, subscribe } from '@wordpress/data';
+/**
+ * Internal dependencies
+ */
+import { createReducerFromActionMap } from '../data/store-helpers';
+
+/**
+ * Register structure store and subscribe to block editor save.
+ *
+ * @param {Object}   opts
+ * @param {string}   opts.storeName   Name of store.
+ * @param {Function} opts.getEndpoint REST API endpoint.
+ * @param {Function} opts.saveError   Handler for displaying errors.
+ * @param {Function} opts.clearError  Handler for clearing errors.
+ * @param {Function} opts.updateBlock Update block with given structure.
+ * @param {Function} opts.readBlock   Extract structure from block.
+ */
+export function registerStructureStore( {
+	storeName,
+	getEndpoint,
+	saveError,
+	clearError,
+	updateBlock,
+	readBlock,
```

Comment:
Maybe rename this to `readBlockStructure`?

Reply:
Yeah that might be better. I ended up moving this around some more in #3958, I can rename in that one to avoid conflicts.